### ISSUE TITLE
sslyze 1.4.2 installed via homebrew causes "certinfo: TypeError - object of type 'UnrecognizedExtension' has no len()" on some sites

### ISSUE THREAD

**Issue by stefanb** [Homebrew/homebrew-core#30489]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [X] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [X] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X] ran `brew update` and can still reproduce the problem?
- [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  docker
  nginx
  intltool
  wireshark
```
These are likely not related, but can link them if needed.
- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
```
$ brew gist-logs sslyze
Error: No logs.
```
- [X] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?
```
$ brew config
HOMEBREW_VERSION: 1.7.1-25-g478f50f
ORIGIN: https://github.com/Homebrew/brew
HEAD: 478f50f589e79728e2e4c00b22ed4af101cd099f
Last commit: 10 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 977fc665ba61fb16566b3d29ac4b0764cf382bce
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
CPU: octa-core 64-bit kabylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 9.1 build 902
Git: 2.15.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 1.8.0_131
macOS: 10.13.6-x86_64
CLT: 10.0.0.0.1.1527767617
Xcode: 9.4.1
XQuartz: 2.7.11 => /opt/X11
```

To help us debug your issue please explain:
- What you were trying to do (and why)
Tried to use sslyze https://github.com/nabla-c0d3/sslyze, installing it via formula https://github.com/Homebrew/homebrew-core/blob/master/Formula/sslyze.rb
- What happened (include command output)
Sslyze is installed without a problem but `sslyze --certinfo` failed on some websites, but worked on the others
- What you expected to happen
I expect it to work on all sites
- Step-by-step reproduction instructions (by running `brew install` commands)
```
brew install sslyze
sslyze --certinfo letsencrypt.org
```

There are numerous reports (currently 7) of sslyze encountering a problem UnrecognizedExtension https://github.com/nabla-c0d3/sslyze/search?q=UnrecognizedExtension&type=Issues

Some users have resorted to installing the sslyze via `pip`, skipping homebrew entirely.

The upstream ticket where suspicion of problematic dependancies/resources is brought up:
https://github.com/nabla-c0d3/sslyze/issues/322

Last relevant changes were done as part of https://github.com/Homebrew/homebrew-core/pull/28332  :
* https://github.com/Homebrew/homebrew-core/commit/f19782edf2694bbec61fdddd4e0295680f9632e7 (by @ilovezfs and @mistydemeo)
* https://github.com/Homebrew/homebrew-core/commit/52a167e08869396f734567550c0b1a877df05fc4 (by @BrewTestBot and @mistydemeo)

I suggest to change the test in formula 
https://github.com/Homebrew/homebrew-core/blob/master/Formula/sslyze.rb#L139
to include this case, eg:
```
  test do
    assert_match "SCAN COMPLETED", shell_output("#{bin}/sslyze --regular google.com")
    assert_not_match "exception", shell_output("#{bin}/sslyze --certinfo letsencrypt.org")
  end
```

**Comment by stefanb** [comment]
This was partially fixed (to prevent the complete crash) in https://github.com/nabla-c0d3/sslyze/issues/320
Further improvement to give a better explanation could be done with changes in the upstream components, which is being tracked in https://github.com/nabla-c0d3/sslyze/issues/323

Within the homebrew project the possible solutions are:
* to adjust the sslyze formula to use sslyze version with thee fixes (presumably 1.4.3, once it is released) and/or
* to ensure that a newer version of openssl is available and used by sslyze - not sure how/if this can be specified within brew formula

**Comment by ilovezfs** [comment]
This should now be fixed by https://github.com/Homebrew/homebrew-core/pull/30794 if you `brew upgrade sslyze`.
